lib/ssz_ex.ex

```diff
@@ -72,6 +88,21 @@ defmodule LambdaEthereumConsensus.SszEx do
     |> flatten_results_by(&Enum.join/1)
   end
 
+  defp encode_bitlist(bit_list, max_size) do
+    len = bit_size(bit_list)
+
+    if len > max_size do
+      {:error, "excess bits"}
+    else
+      r = rem(len, @bits_per_byte)
+      <<pre::bitstring-size(len - r), post::bitstring-size(r)>> = bit_list
+      {:ok, <<pre::bitstring, 1::size(@bits_per_byte - r), post::bitstring>>}
+    end
+  end
+
+  defp encode_bitvector(bit_vector, size) when bit_size(bit_vector) == size, do: {:ok, bit_vector}
```

Comment:
Instead of `{:ok, bit_vector}` here let's return `{:ok, BitVector.to_bytes(bit_vector)}`, which is a fix that has been added a couple of weeks ago (`BitVector.new()` was modified too).

Reply:
there is a warning about the opaque type:
`Call does not have expected opaque term of type LambdaEthereumConsensus.Utils.BitVector.t() in the 1st position` :thinking:

Reply:
Can we add a `BitVector.bit_size` guard in the `BitVector` and use it here to see if the issue gets fixed? On guards: https://hexdocs.pm/elixir/main/Kernel.html#defguard/1